docs/index.rst

```diff
@@ -27,123 +27,316 @@ EdgeDB TypeScript/JS Client
+    type Movie {
+      required property title -> str {
+        constraint exclusive;
+      };
+      multi link actors -> Person;
+    }
+
+    type Person {
+      required property name -> str;
+    }
+  }
+
+Generate the new migration and apply them:
+
+.. code-block:: bash
+
+  $ edgedb migration create
+  $ edgedb migrate
+
+We'll be using TypeScript and Node for this example, so let's setup a simple
+app:
+
+.. code-block:: bash
+
+  $ npm init -y # initialize a new npm project
+  $ npm i edgedb
+  $ npm i -D typescript @types/node @edgedb/generate tsx
+  $ npx tsc --init # initialize basic a basic TypeScript project
+
+Client
+^^^^^^
 
 The ``Client`` class implements the core functionality required to establish a
 connection to your database and execute queries. If you prefer writing queries
 as strings, the Client API is all you need.
 
-.. code-block:: javascript
+Let's create a simple Node.js script that seeds the database by running an
+insert query directly with the driver:
+
+.. code-block:: typescript
+  :caption: seed.ts
+
+  import * as edgedb from "edgedb";
+
+  const client = edgedb.createClient();
+
+  async function main() {
+    await client.execute(`
+      insert Person { name := "Robert Downey Jr." };
+      insert Person { name := "Scarlett Johansson" };
+      insert Movie {
+        title := <str>$title,
+        actors := (
+          select Person filter .name in {
+            "Robert Downey Jr.",
+            "Scarlett Johansson"
+          }
+        )
+      }
+    `, { title: "Iron Man 2" });
+  }
+
+  main();
+
+We can now seed the database by running this script with ``tsx``
+
+.. code-block:: bash
+
+  $ npx tsx seed.ts
+
+Feel free to explore the database in the EdgeDB UI, where you will find the new
+data you inserted through this script, as well as any data you inserted when
+running the Quickstart.
+
+.. note:: A word on module systems
+
+  Different build tools and runtimes have different specifications for how
+  modules are imported, and we support a wide-range of those styles. For
+  clarity, we will be sticking to standard TypeScript-style ESM module importing
+  without a file extension throughout this documentation. Please see your build
+  or environment tooling's guidance on how to adapt this style.
+
+Query
+^^^^^
+
+Now, let's write a Node.js script that queries the database for details about
+Iron Man 2:
+
+.. code-block:: typescript
+  :caption: query.ts
+
+  import * as edgedb from "edgedb";
+
+  const client = edgedb.createClient();
+
+  async function main() {
+    const result = await client.querySingle(`
+      select Movie {
+        title,
+        actors: {
+          name,
+        }
+      } filter .title = "Iron Man 2"
+    `);
+
+    console.log(JSON.stringify(result, null, 2));
+  }
+
+  main();
+
+Interfaces
+^^^^^^^^^^
+
+Since we're using TypeScript, it would be nice to be able to type the return
+value of this query, so let's use our first generator, the :ref:`Queries
```

Comment:
Should `Queries` be capitalized here? Actually, should this maybe be `the interfaces generator` and link to `edgedb-js-interfaces` instead?

Reply:
⚡ [`5b58c22` (#619)](https://github.com/edgedb/edgedb-js/pull/619/commits/5b58c223b934794dd63cb329d8135b1f4dea9363)